astropy/units/format/cds.py

```diff
@@ -275,15 +276,7 @@ def _get_unit(cls, t: LexToken) -> UnitBase:
 
     @classmethod
     def _parse_unit(cls, unit: str, detailed_exception: bool = True) -> UnitBase:
-        if unit not in cls._units:
-            if detailed_exception:
-                raise ValueError(
-                    f"Unit '{unit}' not supported by the CDS SAC standard. "
-                    f"{did_you_mean(unit, cls._units)}"
-                )
-            else:
-                raise ValueError()
-
+        cls._validate_unit(unit, detailed_exception)
```

Comment:
Funny how this now becomes just like `fits` and `ogip` - probably room for a later refactor of `generic` so that this can be removed in all three.

Reply:
In this pull request I'd prefer to only make changes to `CDS`, but in future pull requests it will indeed be possible to reduce code duplication in all the `Generic` subclasses.